osf/models/registrations.py

```diff
@@ -1465,11 +1466,14 @@ def sync_internet_archive_attributes(sender, instance, **kwargs):
             'moderation_state'
         }.intersection(instance.get_dirty_fields().keys())
         current_fields = {key: str(getattr(instance, key)) for key in internet_archive_metadata}
-        if current_fields and instance.is_public:
+        if current_fields and (instance.is_public or current_fields.get('moderation_state') ==  RegistrationModerationStates.WITHDRAWN.db_name):
             requests_retry_session().post(
                 f'{settings.OSF_PIGEON_URL}metadata/{instance._id}',
                 json=current_fields
             )
+    elif settings.IA_ARCHIVE_ENABLED:
+        enqueue_postcommit_task(send_to_pigeon, (instance._id,), {}, celery=True)
```

Comment:
this is going to create an IA entry for every single registration as soon as it's saved instead of waiting for it to be public, no?